### ISSUE TITLE
[a11y]: Dropdown `titleText` prop should be marked as required

### ISSUE THREAD

**Issue by samrobbins85** [carbon-design-system/carbon#12809]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.19.0

### React version

17.0.0

### Automated testing tool and ruleset

Axe Core

### Description

The `aria-labelledby` of the main `button` of the dropdown has two items, e.g. `downshift-11-label downshift-11-toggle-button` The first `downshift-11-label` is always included, but the item with that `id` is only shown when `titleText` is provided, so when `titleText` isn't provided, this reference is broken

### Reproduction/example

https://stackblitz.com/edit/github-ynms4h?file=src/App.jsx

### Steps to reproduce

Inspect the dropdown and see that the `aria-labelledby`'s first item doesn't have a matching `id`

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tw15egan** [comment]
I believe the `titleText` prop is needed for a11y purposes. 
<img width="675" alt="Screenshot 2022-12-07 at 9 05 34 AM" src="https://user-images.githubusercontent.com/11928039/206199840-d953c8fc-935b-46a4-b00b-cafa3d31fec9.png">


If you would like to hide the label, you still need to provide the `titleText` prop, but you can also pass in `hideLabel`

**Comment by samrobbins85** [comment]
Thanks, the `titleText` prop is [optional](https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/Dropdown/Dropdown.js#L394) though, which means that it's difficult to know that you break accessibility by not providing it

**Comment by tw15egan** [comment]
You're correct; we should update this to be required. Would that solve the issue/meet expectations?

**Comment by samrobbins85** [comment]
That would be great thanks
